packages/astral/tests/NodeFinder/SimpleNodeFinderTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\Astral\Tests\NodeFinder;
+
+use Iterator;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassLike;
+use Rector\Core\PhpParser\Parser\SimplePhpParser;
+use Symplify\Astral\Naming\SimpleNameResolver;
+use Symplify\Astral\NodeFinder\SimpleNodeFinder;
+use Symplify\Astral\PhpParser\SmartPhpParser;
+use Symplify\Astral\Tests\HttpKernel\AstralKernel;
+use Symplify\Astral\ValueObject\AttributeKey;
+use Symplify\PackageBuilder\Testing\AbstractKernelTestCase;
+
+final class SimpleNodeFinderTest extends AbstractKernelTestCase
+{
+    /**
+     * @var Node[]
+     */
+    private array $nodes = [];
+
+    private SimpleNodeFinder $simpleNodeFinder;
+
+    protected function setUp(): void
+    {
+        $this->bootKernel(AstralKernel::class);
+        $this->simpleNodeFinder = $this->getService(SimpleNodeFinder::class);
+
+        $phpParser = $this->getService(SmartPhpParser::class);
```

Comment:
I am wondering why the SmartPhpParser does not use `ParentConnectingVisitor ` and `NodeConnectingVisitor ` .. is this intentional?

if so, is there another parser I could use in this test which is configured for the visitors?

see https://github.com/nikic/PHP-Parser/pull/681/files

Reply:
Feel free to add them.